src/xtgeo/grid3d/_gridprop_roxapi.py

```diff
@@ -176,16 +178,25 @@ def _store_in_roxar(self, pname, roxgrid, realisation, casting):  # pragma: no c
         rprop.code_names = self.codes.copy()
 
 
-def _fix_codes(codes):  # pragma: no cover
-    """Roxar may provide a code list with empty strings; fix this issue here."""
+def _fix_codes(self, codes):  # pragma: no cover
+    """Roxar may provide a code list with empty strings values, fix this issue here.
+
+    Roxar may also interpolate code values which are actually not present in the
+    property. Here, the presence of actual codes is also checked.
+    """
     newcodes = {}
+    codes_data = {val: str(val) for val in np.unique(self.get_active_npvalues1d())}
+
     for code, name in codes.items():
         if not isinstance(code, int):
             code = int(code)
 
         if not name:
             name = str(code)
 
+        if code not in codes_data.keys():
```

Comment:
As the codes_data dictionary is not used for anything else, it would be better to skip generating it and just check against the active values here. Something like:
`if code not in self.get_active_npvalues1d():`